server/server.go

```diff
@@ -250,3 +258,10 @@ func WithProfilerEndpoint() Option {
 		s.pprofEnabled = true
 	}
 }
+
+// TODO: add comment
```

Comment:
```suggestion
// WithServerConfig is an option function that overrides the default configuration settings of the server.
// This can give the client more control over the server feature functions
```